runtime/Tracer.cpp

```diff
@@ -0,0 +1,113 @@
+
+#include "Tracer.h"
+#include "Shadow.h"
+#include "llvm/Support/raw_ostream.h"
+
+nlohmann::json Tracer::currentTrace;
+nlohmann::json Tracer::expressions;
+nlohmann::json Tracer::pathConstraints;
+const std::string Tracer::BACKEND_TRACE_FILE = "/tmp/backend_trace.json";
```

Comment:
This should be either configurable or created in more generic way (e.g. `mktemp`).